teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,465 @@
+     experimentation needs to be done around this
+
+   The next action are:
+   - Mark this TEP as withdrawn, we acknoledge it is not the way to
+     go. When marking this as `withdrawn`, add the reason why.
+   - Conduct experiment on not using PipelineResource
+   - Act that the `PipelineResource` needs a full re-design (and thus
+     removing it from the beta API for now)
+4. From the conducted experiment on "a life without PipelineResource",
+   two concept are being discussed:
+   - Workspace : to share data between tasks
+   - Results : to share results between tasks
+
+   A TEP for each is created, approved and implemented.
+5. Design discussion and docs are being created to re-design
+   PipelineResources using the above new concept. It gets discussed in
+   different working group.
+   A TEP is created to act that design work is going on on the
+   subject. The TEP is marked as `proposed`.
+6. A design is agreed after working group discussions, this new TEP
+   gets updated, and is marked as `implementable`.
+7. Work can start on the new `PipelineResource` design
+8. Once the work around this is done, the TEP gets updated to
+   `implemented` state.
+
+Later, some enhancement to the `PipelineResource` are proposed. Those
+will result in new `TEP`s.
+
+## Drawbacks
+
+Any additional process has the potential to engender resentment within
+the community. There is also a risk that the TEP process as designed
+will not sufficiently address the scaling challenges we face today. PR
+review bandwidth is already at a premium and we may find that the TEP
+process introduces an unreasonable bottleneck on our development
+velocity.
+
+It certainly can be argued that the lack of a dedicated issue/defect
+tracker beyond GitHub issues contributes to our challenges in managing
+a project as large as Tekton, however, given that other large
+organizations, including GitHub itself, make effective use of GitHub
+issues, perhaps the argument is overblown.
+
+The centrality of Git and GitHub within the TEP process also may place
+too high a barrier to potential contributors, however, given that both
+Git and GitHub are required to contribute code changes to Tekton today
+perhaps it would be reasonable to invest in providing support to those
+unfamiliar with this tooling. It also make the proposal document more
+accessible that what it is before this proposal, as you are required
+to be part of
+[tekton-users@](https://groups.google.com/forum/#!forum/tekton-users)
+or [tekton-dev@](https://groups.google.com/forum/#!forum/tekton-dev)
+google groups to see the design docs.
+
+## Alternatives
+
+This TEP process is related to
+- the generation of a architectural roadmap
+- the fact that the what constitutes a feature is still undefined
+- issue management
+- the difference between an accepted design and a proposal
+- the organization of design proposals
+
+This proposal attempts to place these concerns within a general
+framework.
```

Comment:
i dont really understand this section - do we want to add more detail to flesh out the above points?

i also thought this section might touch on some of the alternatives we discussed but it doesnt seem to have any alternatives?

Reply:
right, I need to rework that one